src/Wallet.purs

```diff
@@ -174,30 +100,110 @@ dummySign tx@(Transaction { witnessSet: tws@(TransactionWitnessSet ws) }) =
           "ed25519_sig1ynufn5umzl746ekpjtzt2rf58ep0wg6mxpgyezh8vx0e8jpgm3kuu3tgm453wlz4rq5yjtth0fnj0ltxctaue0dgc2hwmysr9jvhjzswt86uk"
     )
 
+-------------------------------------------------------------------------------
+-- Gero backend
+-------------------------------------------------------------------------------
+
+mkGeroWalletAff :: Aff Wallet
+mkGeroWalletAff = do
+  gero <- enable "gerowallet"
+  -- Ensure the Gero wallet has collateral set up
+  whenM (isNothing <$> getCollateral gero)
+    (liftEffect $ throw "Gero wallet missing collateral")
+  connection <- liftEffect $ Ref.new gero
+  pure $ Wallet
+    { connection
+    , getWalletAddress
+    , getCollateral
+    , signTx
+    , signTxBytes
+    }
+
+-------------------------------------------------------------------------------
+-- Helper functions
+-------------------------------------------------------------------------------
+
+enable :: String -> Aff Cip30Connection
```

Comment:
I would prefer to avoid a stringly-typed identifier for different wallets, since there are really only two options we can support at the moment

Reply:
We could have `data WalletApi = Nami | Gero` or something and just convert it to a string